src/otaclient/main.py

```diff
@@ -146,42 +214,8 @@ def main() -> None:  # pragma: no cover
     except Exception as e:
         logger.warning(f"failed to read system uptime: {e}")
 
-    if _env.is_dynamic_client_preparing():
-        logger.info("preparing downloaded dynamic ota client ...")
-        try:
-            logger.info("mounting dynamic client squashfs ...")
-            client_package_prepareter = OTAClientPackagePreparer(
-                squashfs_file=cfg.DYNAMIC_CLIENT_SQUASHFS_FILE,
-                mount_base=cfg.DYNAMIC_CLIENT_MNT,
-                active_root=cfg.ACTIVE_ROOT,
-                active_slot_mnt_point=cfg.ACTIVE_SLOT_MNT,
-                host_root_mnt_point=cfg.DYNAMIC_CLIENT_MNT_HOST_ROOT,
-                bootloader=ecu_info.bootloader,
-            )
-            client_package_prepareter.mount_client_package()
-
-            _mount_base = cfg.DYNAMIC_CLIENT_MNT
-            logger.info(f"changing root to {_mount_base}")
-            os.chroot(_mount_base)
-            os.chdir("/")
-
-            logger.info("execve for dynamic client runnning ...")
-            running_env = os.environ.copy()
-            del running_env[cfg.PREPARING_DOWNLOADED_DYNAMIC_OTA_CLIENT]
-            running_env[cfg.RUNNING_DOWNLOADED_DYNAMIC_OTA_CLIENT] = "yes"
-            # the process should finish after this execve call
-            DYNAMIC_CLIENT_PYTHON_PATH = "/otaclient/venv/bin/python3"
-            os.execve(
-                path=DYNAMIC_CLIENT_PYTHON_PATH,
-                argv=[DYNAMIC_CLIENT_PYTHON_PATH, "-m", "otaclient"],
-                env=running_env,
-            )
-        except Exception as e:
-            logger.exception(f"Failed during dynamic client preparation: {e}")
-        return sys.exit(1)
-
-    if not _env.is_dynamic_client_running():
-        # in dynamic client, the pid file has already been created
+    # for dynamic loaded otaclient, skip other otaclient check
+    if not _env.is_running_as_downloaded_dynamic_app():
```

Comment:
OK, so when the new app otaclient managed by systemd is launched, a new process will be created.
- `sudo systemctl restart otaclient`
→ This restarts the original service. What will happen to the app otaclient in this case?
- `sudo systemctl restart otaclient.service_app`
→ This restarts the app service, correct? And basically, this command should not be used.